tests/e2e/vmops_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite contains tests to verify VM operations (snapshot) with
+// vSphere docker volumes attached to the VM.
+
+// +build runonce
+
+package e2e
+
+import (
+	"log"
+	"fmt"
+	adminutils "github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	vimCmd = "vim-cmd "
+
+	// VMId get the VM ID from ESX host
+	VMId = vimCmd + "vmsvc/getallvms|grep %s|cut -f 1 -d\" \""
+	// VMTakeSnapshot - take a snap of a VM
+	VMTakeSnapshot = vimCmd + "vmsvc/snapshot.create %s"
+	// VMRemoveSnapshots - delete all snaps of a VM
+	VMRemoveSnapshots = vimCmd + "vmsvc/snapshot.removeall %s"
+)
+
+// VMOpsTest - struct for vmgroup tests
+type VMOpsTest struct {
+	config        *inputparams.TestConfig
+	testContainer string
+	volName1      string
+	volName2      string
+}
+
+var _ = Suite(&VMOpsTest{})
+
+func (vm *VMOpsTest) SetUpSuite(c *C) {
+	vm.config = inputparams.GetTestConfig()
+	if vm.config == nil {
+		c.Skip("Unable to retrieve test config, skipping vmgroup tests")
+	}
+	adminutils.ConfigInit(vm.config.EsxHost)
+
+	vm.testContainer = inputparams.GetUniqueContainerName("Test_VMOps")
+
+	vm.volName1 = inputparams.GetUniqueVolumeName("Test_VMOps")
+	vm.volName2 = inputparams.GetUniqueVolumeName("Test_VMOps")
+
+	// Create the test volumes
+	_, err := dockercli.CreateVolumeWithOptions(vm.config.DockerHosts[0], vm.volName1, "-o attach-as=persistent")
+	c.Assert(err, IsNil, Commentf("Failed creating volume %s\n", vm.volName1))
+
+	_, err = dockercli.CreateVolume(vm.config.DockerHosts[0], vm.volName2)
```

Comment:
minor: I believe default is `independent` .. isn't it good idea to create volume with `"-o attach-as=independent` for readability as well as none of our test creates volume with such option explicitly?